src/main/java/org/mule/extension/file/internal/lock/FileChannelPathLock.java

```diff
@@ -105,4 +149,17 @@ public void release() {
   public Path getPath() {
     return path;
   }
+
+  private void sleepThread(long timeout) {
+    if (timeout <= 0) {
+      return;
+    }
+    try {
+      Thread.sleep(TimeUnit.NANOSECONDS.toMillis(timeout) / 20L);
+    } catch (InterruptedException e) {
```

Comment:
This is highly debatable. There will always be corner cases. If the file is requested so much in that 10 minutes time frame, the user should reconsider the architecture o increase the lock timeout.
If we put a max sleep time, we come closer and closer to straight up busy wait, especially for longer timeouts, which is what we wanted to avoid in the first place. I think 20 retries is a good compromise.